### ISSUE TITLE
GPU Not Utilized for WebGL Rendering

### ISSUE THREAD

**Issue by hassanaleem** [Alex313031/thorium#345]
**System Details**
 - OS:  Ubuntu 22.04
 - Thorium Version 117.0.5938.157 stable

**Problem**
The dedicated GPU is not effectively utilized for WebGL rendering. This issue is observed on specific webpages, such as "https://thorium.rocks/thoriumos" and pages with more complex 3D models like "(https://threejs.org/examples/#webgl_animation_keyframes)."

**Screenshots**
For "https://thorium.rocks/thoriumos":
![Screenshot from 2023-10-23 12-32-15](https://github.com/Alex313031/ThoriumOS/assets/59873682/7b44fe19-14ac-4a17-9be3-c98253bf99cf)

For "(https://threejs.org/examples/#webgl_animation_keyframes)":
![Screenshot from 2023-10-23 12-33-51](https://github.com/Alex313031/ThoriumOS/assets/59873682/0c1c26b0-779f-435b-8d6a-1e23df235401)

**Additional Notes**
This issue has been tested on Ubuntu 22.04 with kernel 6.2.0-35, using Nvidia Driver version 535.113.01 and Cuda Version 12.2.

**Comment by binboupan** [comment]
It runs at 120fps at 1440p for me using a 1650 Super. Are you using Wayland or Xorg? Hardware accelleration is disabled on the former.

**Comment by hassanaleem** [comment]
I am using X11 (Xorg). 


To illustrate this issue, I've conducted a comparison between Thorium Browser, Thorium Browser with a dedicated GPU, and Firefox with a dedicated GPU. (done using "launch with Discrete Graphics card")
Using this website https://threejs.org/examples/?q=perfo#webgl_instancing_performance

https://github.com/Alex313031/thorium/assets/59873682/971286e7-21d6-40fb-8d49-ba6ad625b99d


I've found a temporary solution by changing the Nvidia Prime profile to "performance." This effectively shifts the processing load to the GPU, resulting in improved performance. However, this comes at the cost of increased power consumption, which may not be ideal for all users. 

Ideally, the system should be able to automatically utilize the GPU when required, without the need to set the Nvidia Prime profile to "performance." This would help improve performance without excessively draining power.
